includes/admin/class-wc-admin-setup-wizard.php

```diff
@@ -1466,6 +1493,65 @@ class="jetpack-logo"
 		<?php
 	}
 
+	protected function get_all_activate_errors() {
+		return array(
+			'default' => __( "Sorry! We tried, but we couldn't connect Jetpack just now 😭. Please go to the Plugins tab to connect Jetpack, so that you can finish setting up your store.", 'woocommerce' ),
+			'jetpack_cant_be_installed' => __( "Sorry! We tried, but we couldn't install Jetpack for you 😭. Please go to the Plugins tab to install it, and finish setting up your store.", 'woocommerce' ),
+			'jetpack_cant_be_activated' => __( "Sorry! We tried, but we couldn't activate Jetpack for you 😭. Please go to the Plugins tab to activate it, and finish setting up your store.", 'woocommerce' ),
+			'register_http_request_failed' => __("Sorry! We couldn't contact Jetpack just now 😭. Please make sure that your site is visible over the internet, and that it accepts incoming and outgoing requests via curl. You can also try to connect to Jetpack again, and if you run into any more issues, please contact support.", 'woocommerce'),
+			'siteurl_private_ip_dev' => __( "Whoops! We couldn't connect. Your site is probably on a private network. Jetpack can only connect to public sites. Please make sure your site is visible over the internet, and then try connecting again 🙏." , 'woocommerce'),
+		);
+	}
+
+	protected function get_activate_error_message( $code = '' ) {
+		$errors = $this->get_all_activate_errors();
+		return array_key_exists( $code, $errors ) ? $errors[ $code ] : $errors['default'];
+	}
+
+	protected function install_and_activate_jetpack_sync() {
+		include_once( ABSPATH . 'wp-admin/includes/plugin.php' );
+
+		// If Jetpack is not installed, install it.
+		if ( 0 !== validate_plugin( 'jetpack/jetpack.php' ) ) {
+
+			include_once( ABSPATH . '/wp-admin/includes/admin.php' );
+			include_once( ABSPATH . '/wp-admin/includes/plugin-install.php' );
+			include_once( ABSPATH . '/wp-admin/includes/class-wp-upgrader.php' );
+			include_once( ABSPATH . '/wp-admin/includes/class-plugin-upgrader.php' );
+
+			$api = plugins_api( 'plugin_information', array( 'slug' => 'jetpack' ) );
+
+			if ( is_wp_error( $api ) ) {
+				// TODO: log original error message
+				return 'jetpack_cant_be_installed';
+			}
+			$upgrader = new Plugin_Upgrader( new Automatic_Upgrader_Skin() );
+			$result = $upgrader->install( $api->download_link );
+
+			if ( true === $result ) {
+				// Continue with activation after successful install.
+				$this->install_and_activate_jetpack_sync();
+			} else {
+				// TODO: log original error message
+				return 'jetpack_cant_be_installed';
+			}
+		} else if ( ! class_exists( 'Jetpack' ) ) {
+			$result = activate_plugin( 'jetpack/jetpack2.php' );
```

Comment:
What's the use case for `jetpack2.php`?

Reply:
That's an error - was using this for testing. Fixed & rebased, new version at 85dd98da22967655ea244e55ce5d5c32e52f3e11